File: scratch/test_scrape.js
async function main() {
    try {
        const url = "https://driving-tests.org/cdl-handbook/";
        const res = await fetch(url, {
            headers: {
                "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
            }
        });
        console.log("Status:", res.status);
        const text = await res.text();
        
        // Find links using regex
        const hrefRegex = /href="([^"]+)"/g;
        let match;
        const links = [];
        while ((match = hrefRegex.exec(text)) !== null) {
            const href = match[1];
            if (href.includes("/cdl-handbook") || href.includes("-cdl-") || href.includes("/cdl/")) {
                links.push(href);
            }
        }
        
        console.log(`Found ${links.length} matching links. Sample:`);
        console.log(links.slice(0, 30));
    } catch (err) {
        console.error("Error:", err);
    }
}
main();

File: scratch/test_scrape.py
import requests
from bs4 import BeautifulSoup
import re

url = "https://driving-tests.org/cdl-handbook/"
headers = {
    "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"
}

try:
    response = requests.get(url, headers=headers, timeout=15)
    print("Status:", response.status_code)
    soup = BeautifulSoup(response.text, "html.parser")
    
    # Let's find all links on the page that point to state CDL handbooks
    links = []
    for a in soup.find_all("a", href=True):
        href = a["href"]
        if "/cdl-handbook" in href or "-cdl-" in href:
            links.append((a.get_text(strip=True), href))
            
    print(f"Found {len(links)} links. Sample:")
    for label, href in links[:20]:
        print(f" - {label}: {href}")
        
except Exception as e:
    print("Error:", e)
